docker/daemon.go

```diff
@@ -141,7 +151,17 @@ func mainDaemon() {
 	job.Setenv("TlsCert", *flCert)
 	job.Setenv("TlsKey", *flKey)
 	job.SetenvBool("BufferRequests", true)
-	if err := job.Run(); err != nil {
-		log.Fatal(err)
+	err := job.Run()
+
+	// Wait for the daemon startup goroutine to finish
+	// This makes sure we can actually cleanly shutdown the daemon
+	log.Infof("waiting for daemon to initialize")
+	wg.Wait()
```

Comment:
No, the whole reason this is setup the way it is is so the API can be available to take requests while the daemon is still booting.